red-hat-storage/cephci
# CEPH-9232: Test to verify PG rebalancing during OSD full upon additionon of new OSDs

[CEPH-9232:](https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-9232) Tier-4 test to verify cluster behavior and PG rebalancing when OSDs are full (cluster full) and new OSDs are added to the cluster

Test suites modified -
- suites/pacific/rados/tier-2_rados_test-osd-rebalance.yaml
- suites/pacific/rados/tier-4_rados_test-pool-osd-recovery.yaml
- suites/quincy/rados/tier-2_rados_test-osd-rebalance.yaml
- suites/quincy/rados/tier-4_rados_test-pool-osd-recovery.yaml

Test modules modified - 
- ceph/rados/core_workflows.py
- ceph/rados/utils.py
- tests/cephadm/test_host.py

Test modules added -
- tests/rados/test_osd_full.py

Steps:
1. Create replicated pool with single PG and autoscaling disabled
2. Retrieve the acting set for the pool"
3. Calculate the amount of data to be written to pools in order to completely full OSD health warnings"
4. Perform IOPS using rados bench to fill up the pool up till completely full capacity"
5. Verify the number of objects written to the pool and the cluster"
6. Perform IOPS again and ensure IOPS was unsuccessful as OSDs are full"
7. Add new OSD host and deploy new OSDs"
8. Enable autoscaling and observe the acting set OSD's 'used capacity' decreasing"
10. Eventually cluster full warning should disappear, wait for PGs to achieve active+clean state"
11. Remove the newly add OSD host and wait for PGs to achieve active+clean state"

Logs:
- RHCS 6.1:
  - CEPH-9232: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-8WNJHR
  - CEPH-83571715: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-HVX805

- RHCS 5.3:
  - CEPH-9232: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-WS6CJ8
  - CEPH-83571715: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-0R2BFH

Signed-off-by: Harsh Kumar <hakumar@redhat.com>
